app/src/main/java/org/schabi/newpipe/Downloader.java

```diff
@@ -276,6 +278,36 @@ public DownloadResponse post(String siteUrl, DownloadRequest request) throws IOE
             return null;
         }
 
-        return new DownloadResponse(body.string(), response.headers().toMultimap());
+        return new DownloadResponse(response.code(), body.string(), response.headers().toMultimap());
+    }
+
+    @Override
+    public DownloadResponse head(String siteUrl) throws IOException, ReCaptchaException {
+        final HttpsURLConnection con = (HttpsURLConnection) new URL(siteUrl).openConnection();
+
+        try {
+            con.setRequestMethod("HEAD");
+
+            con.setConnectTimeout(30 * 1000);// 30s
+            con.setReadTimeout(30 * 1000);// 30s
+
+            con.setRequestProperty("User-Agent", USER_AGENT);
+            if (!TextUtils.isEmpty(mCookies)) {
+                con.setRequestProperty("Cookie", mCookies);
+            }
+
+        } catch (Exception e) {
+            /*
+             * HTTP 429 == Too Many Request Receive from Youtube.com = ReCaptcha challenge
+             * request See : https://github.com/rg3/youtube-dl/issues/5138
+             */
+            if (con.getResponseCode() == 429) {
+                throw new ReCaptchaException("reCaptcha Challenge requested", con.getURL().toString());
+            }
+
+            throw new IOException(con.getResponseCode() + " " + con.getResponseMessage(), e);
+        }
+
+        return new DownloadResponse(con.getResponseCode(), null, con.getHeaderFields());
     }
```

Comment:
Even if this is just temporary, we should still use OkHttp's client:

```java
@Override
public DownloadResponse head(String siteUrl) throws IOException, ReCaptchaException {
    final Request request = new Request.Builder()
            .head().url(siteUrl)
            .addHeader("User-Agent", USER_AGENT)
            .build();
    final Response response = client.newCall(request).execute();

    if (response.code() == 429) {
        throw new ReCaptchaException("reCaptcha Challenge requested", siteUrl);
    }

    return new DownloadResponse(response.code(), null, response.headers().toMultimap());
}
```

Reply:
Oh yes, you are right.

Reply:
Done. Thanks for the code :)